pkg/kubectl-argo-rollouts/cmd/create/create.go

```diff
@@ -0,0 +1,279 @@
+	}
+}
+
+// NewCmdCreateAnalysisRun returns a new instance of an `rollouts create analysisrun` command
+func NewCmdCreateAnalysisRun(o *options.ArgoRolloutsOptions) *cobra.Command {
+	createOptions := CreateAnalysisRunOptions{
+		ArgoRolloutsOptions: *o,
+	}
+	var cmd = &cobra.Command{
+		Use:   "analysisrun",
+		Short: "Create an AnalysisRun from a template",
+		Example: o.Example(`
+  # Create an AnalysisRun from a local template file
+  %[1]s create analysisrun --from-file my-analysis-template.yaml
+  # Create an AnalysisRun from a template in the cluster
+  %[1]s create analysisrun --from my-analysis-template
+`),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			froms := 0
+			if createOptions.From != "" {
+				froms++
+			}
+			if createOptions.FromFile != "" {
+				froms++
+			}
+			if froms != 1 {
+				return fmt.Errorf("one of --from or --from-file must be specified")
+			}
+			templateArgs, err := createOptions.ParseArgs()
+			if err != nil {
+				return err
+			}
+			ns := o.Namespace()
+			var template *v1alpha1.AnalysisTemplate
+			if createOptions.From != "" {
+				template, err = createOptions.RolloutsClientset().ArgoprojV1alpha1().AnalysisTemplates(ns).Get(createOptions.From, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+			} else {
+				fileBytes, err := ioutil.ReadFile(createOptions.FromFile)
+				if err != nil {
+					return err
+				}
+				var tmpl v1alpha1.AnalysisTemplate
+				err = unmarshal(fileBytes, &tmpl)
+				if err != nil {
+					return err
+				}
+				template = &tmpl
+			}
+			var name, generateName string
+			if createOptions.Name != "" {
+				name = createOptions.Name
+			} else if createOptions.GenerateName != "" {
+				generateName = createOptions.GenerateName
+			} else {
+				generateName = template.Name + "-"
+			}
+
+			run, err := analysisutil.NewAnalysisRunFromTemplate(template, templateArgs, name, generateName, ns)
+			if err != nil {
+				return err
+			}
+			created, err := createOptions.RolloutsClientset().ArgoprojV1alpha1().AnalysisRuns(ns).Create(run)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(createOptions.Out, "analysisrun.argoproj.io/%s created\n", created.Name)
+			return nil
+		},
+	}
+	o.AddKubectlFlags(cmd)
+	cmd.Flags().StringVar(&createOptions.Name, "name", "", "Use the specified name for the run")
+	cmd.Flags().StringVar(&createOptions.GenerateName, "generate-name", "", "Use the specified generateName for the run")
+	cmd.Flags().StringArrayVarP(&createOptions.ArgFlags, "argument", "a", []string{}, "Arguments to the parameter template")
+	cmd.Flags().StringVar(&createOptions.From, "from", "", "Create an AnalysisRun from an AnalysisTemplate in the cluster")
+	cmd.Flags().StringVar(&createOptions.FromFile, "from-file", "", "Create an AnalysisRun from an AnalysisTemplate in a local file")
+	return cmd
+}
+
+func (c *CreateAnalysisRunOptions) ParseArgs() ([]v1alpha1.Argument, error) {
```

Comment:
Nix: I thought this function was parsing command line args instead of the args for the analysisRun when I saw it earlier in the code. Can we rename this to `ParseAnalysisArgs` to make it more clear?

Reply:
I guess the `templateArgs` in `templateArgs, err := createOptions.ParseArgs()` makes that more clear, but I didn't connect those dots when I looked at it the first time.

Reply:
Actually it is parsing CLI args and returning template args, but I can rename